deploy/README.md

```diff
@@ -0,0 +1,56 @@
+# Deployment
+
+This directory contains scripts for deploying the Workbench API/UIs. To deploy
+individual services, see [../api/project.rb](../api/project.rb) `deploy-api` and
+`deploy-public-api` and [../ui/project.rb](../ui/project.rb) `deploy-ui`. Test
+and staging are automatically pushed by CircleCI.
+
+- test: pushed by Circle on every master branch merge
+- staging: pushed by Circle when someone tags a new release
+- stable: manually pushed using a tagged staging release
+
+The scripts in this directory are to be used for manual deployment to stable
+(and later, prod). Ideally we'd do this in some cloud service, but none of the
+standard solutions have been cleared by security for use; for now we must run
+these processes locally.
+
+## Deploy to staging
+
+- Verify that the most recent master commit [passed on Circle](https://circleci.com/gh/all-of-us/workflows/workbench/tree/master)
+- Go to the [GitHub releases page](https://github.com/all-of-us/workbench/releases);
+  note the latest version (e.g. v0-1-rc1)
+- Click "Draft new release"
+- Increment the minor version and reset the RC/patch version from the previous
+  release (e.g. if the previous were "v0-2-rc5", next would be "v0-3-rc1")
+- Wait for the staging release to complete on Circle
+
+## Deploy to stable
+
+- Take the latest staging release version; ensure that Circle staging deployment
+  succeeded.
+- Run deployment:
+  ```
+  deploy$ ./project.rb --project all-of-us-rw-stable \
+    --account ACCOUNT --version VERSION_TAG --promote
```

Comment:
For RDR we by default grab the version from whatever is running in staging (and then prompt the user to confirm)... makes it less likely somebody will mess up and enter the wrong thing. We could consider doing that here, too, at some point.

If you want to deploy your changes from a development branch to a cloud environment (e.g. a sandbox environment in future) will that be possible with this approach?

Reply:
1. Good idea. I had a less useful TODO in deploy.rb; will add the default version logic in this PR.
2. It's possible, but maybe not as straightforward. Would it be acceptable for this use case if we required that you push your changes to GitHub (either master or minimally a topic branch) as a preqrequisite to doing this kind of deployment? Note: the individual ui/project.rb and api/project.rb deploy commands would still be usable against your locally modified client.

Reply:
Yeah, that sounds fine.